src/worker/ingestion/process-event.ts

```diff
@@ -324,10 +281,55 @@ export class EventsProcessor {
         return returnedProps
     }
 
+    private async setIsIdentified(teamId: number, distinctId: string, isIdentified = true): Promise<void> {
+        let personFound = await this.db.fetchPerson(teamId, distinctId)
+        if (!personFound) {
+            try {
+                personFound = await this.db.createPerson(
+                    DateTime.utc(),
+                    {},
+                    teamId,
+                    null,
+                    true,
+                    new UUIDT().toString(),
+                    [distinctId]
+                )
+            } catch {
+                // Catch race condition where in between getting and creating,
+                // another request already created this person
+                personFound = await this.db.fetchPerson(teamId, distinctId)
+            }
+        }
+        if (personFound && !personFound.is_identified) {
+            await this.db.updatePerson(personFound, { is_identified: isIdentified })
+        }
+    }
+
+    private async handleIdentifyOrAlias(
+        event: string,
+        properties: Properties,
+        distinctId: string,
+        teamId: number
+    ): Promise<void> {
+        if (isDistinctIdIllegal(distinctId)) {
+            this.pluginsServer.statsd?.increment(`illegal_distinct_ids.total`, { distinctId })
+            return
+        }
+        if (event === '$create_alias') {
+            await this.alias(properties['alias'], distinctId, teamId, false)
```

Comment:
ah yes there's a simple way to do this, will update

Reply:
& thoughts on the testing part?

> Maybe it would be good if we had tests for all 8 cases that you outlined awesomely for both functions (create_alias and identify) separately & checking that we end up with the right people & right identified state. It will be a bit of duplication in tests, but I think it'll be harder to break things in the future.